### ISSUE TITLE
LSTM multiple inputs and one input

### ISSUE THREAD

**Issue by ai559031** [keras-team/keras#13086]
Hi, I am trying to build a multiple inputs LSTM model, I expect to use price and sentiment to predict the future price, and since I assume the price for time t will be affected by previous 51 hours' price, I have one def function to help me look back 51 hours:

```
def create_dataset(dataset, look_back):
    dataX, dataY = [], []
    for i in range(len(dataset) - look_back):
        a = dataset[i:(i + look_back), 0]
        dataX.append(a)
        dataY.append(dataset[i + look_back, 0])
    print(len(dataY))
    return np.array(dataX), np.array(dataY)
```

And here is my model:  
```
import keras
def buildModel(dataLength, labelLength):
    price=Input(shape=(dataLength, 51),name='price') 
#     price = Input(shape = (dataLength,1),name='price')
    sentiment = Input(shape=(dataLength, 51),name='sentiment')
    
    priceLayers = LSTM(64, return_sequences=False)(price)
    sentimentLayers = LSTM(64, return_sequences=False)(sentiment)
    
    output = keras.layers.concatenate(
        [priceLayers,sentimentLayers,]
    )
    output = Dense(labelLength, activation='linear',name='output')(output)
    
    model = Model(
        inputs = [price,sentiment],
        outputs=[output]
    )
    model.compile(optimizer='rmsprop',loss='mse')
    return model
```
My size of the train and test data shape: 
trainX.shape = (22234, 51, 1)
testX.shape =(9500, 51, 1)

trainY.shape = (22234,)
testY.shape = (9500,)

trainS.shape = (22234, 51, 1)
testS.shape = (9500, 51, 1)

But when I try to put my dataset into the model and fit: 
```
lstm = buildModel(22234,51)
lstm.fit([trainX,trainS],[trainY],validation_data=(
        [testX,testS],[testY]),epochs = 10
)
```
I got error: 
```
<ipython-input-125-01036d60b144> in <module>()
      4 lstm = buildModel(22234,51)
      5 lstm.fit([trainX,trainS],[trainY],validation_data=(
----> 6         [testX,testS],[testY]),epochs = 10
      7 )
```
`Error when checking input: expected price to have shape (22234, 51) but got array with shape (51, 1)
`
I don't understand which step goes wrong...

**Comment by kushalj001** [comment]
@ai559031 you are passing dataLength as 22234 which, if I am right, is the number of training examples in your dataset. This results in shape in the Input layer to be (22234,51) which is not correct. The input_shape should reflect the size of a single vector in your training set and not the batch size. 
Try passing shape as (51,1) only.
From the docs:

![image](https://user-images.githubusercontent.com/32245327/61034917-ccc6e000-a3e3-11e9-8ca2-e3cc5dfe68f4.png)

Also, it would be better if you reshape your labels into an array of dimensions (22234,1) instead of just (22234, ).

**Comment by gowtham-kp** [comment]
@ai559031 Did that answer your question. Can I close the issue?
